pip/req/req_set.py

```diff
@@ -439,6 +461,11 @@ def _prepare_file(self, finder, req_to_install):
             # # vcs update or unpack archive # #
             # ################################ #
             if req_to_install.editable:
+                if require_hashes:
+                    raise InstallationError(
+                        'The editable requirement %s cannot be installed when '
+                        'requiring hashes, because there is no single file to '
+                        'hash.' % req_to_install)
```

Comment:
It's difficult to cram a yak into a rabbit hole, so, when I find one, I appreciate the engineering effort. ;-)